src/clojure_tic_tac_toe/input_output.clj

```diff
@@ -1,13 +1,67 @@
 (ns clojure-tic-tac-toe.input_output
   (:require [clojure.string :as str]
             [clojure-tic-tac-toe.board :as board]
+            [clojure-tic-tac-toe.computer_move :as computer_move]
             [clojure-tic-tac-toe.utilities :refer [join-lines]]))
 
 (defn display-introduction []
   (println "This is a Tic Tac Toe program.\n"))
 
-(defn display-instructions []
-  (println (join-lines ["To enter a move, type a number from 1-9."
+(defn display-game-mode-instructions []
+  (println (join-lines ["Please enter one of the following:"
+                        "- \"h\" to play another person"
+                        "- \"c\" to play against a computer\n"])))
+
+
+(def valid-game-modes #{:h :c})
+
+(defn- display-angle-bracket []
+  (print "> ")
+  (flush))
+
+(defn- get-input []
+  (keyword (str/trim (read-line))))
+
+(defn- is-game-mode-invalid?
+  [mode]
+  (not (contains? valid-game-modes mode)))
+
+(defn- display-invalid-game-mode-message
+  [mode]
+  (printf "\n<!> Error: Game mode \"%s\" is invalid.", (name mode))
+  (printf " Must be a choice from above."))
+
+(defn- prompt-player-for-game-mode []
+  (printf "\nPlease enter the game mode: "))
+
+(defn get-game-mode []
+  (display-angle-bracket)
+  (loop []
+    (let [mode (get-input)]
+      (if (is-game-mode-invalid? mode)
+        (do
+          (display-invalid-game-mode-message mode)
+          (prompt-player-for-game-mode)
+          (flush)
+          (recur))
+        mode))))
```

Comment:
regarding keeping is-game-mode-invalid? inside of here:

I can think of two things you can do:
1. create another namespace for input validations
2. make this function return the input value, and keep the loop at a higher level.

I'm thinking you can even do a combination of both things. Maybe you can create a function that does both given a validator and the "get" function.

Reply:
In general, I would try and avoid putting too much logic other than just inputting and outputing in this namespace

Reply:
I don't quite understand what you mean in point 2. above ("make this function..."). Do you mean to just have `get-game-mode` call `get-input`, then have a separate function that loops, and calls `get-game-mode`? Because that wouldn't really help...

Reply:
I've cleaned up `input_output.clj`, so it doesn't depend on `board` anymore (instead, a couple of its functions depend on `view`, which in turn depends on `board`.